apis/v1alpha2/gateway_types.go

```diff
@@ -374,59 +376,33 @@ const (
 	TLSModePassthrough TLSModeType = "Passthrough"
 )
 
-// RouteBindingSelector defines a schema for associating routes with the Gateway.
-// If Namespaces and Selector are defined, only routes matching both selectors are
-// associated with the Gateway.
-type RouteBindingSelector struct {
-	// Namespaces indicates in which namespaces Routes should be selected
-	// for this Gateway. This is restricted to the namespace of this Gateway by
+// ListenerRoutes defines which Routes may be attached to this Listener.
+type ListenerRoutes struct {
+	// Namespaces indicates which namespaces Routes may be attached to this
+	// Listener from. This is restricted to the namespace of this Gateway by
 	// default.
 	//
 	// Support: Core
 	//
 	// +optional
 	// +kubebuilder:default={from: Same}
 	Namespaces *RouteNamespaces `json:"namespaces,omitempty"`
-	// Selector specifies a set of route labels used for selecting
-	// routes to associate with the Gateway. If this Selector is defined,
-	// only routes matching the Selector are associated with the Gateway.
-	// An empty Selector matches all routes.
-	//
-	// Support: Core
-	//
-	// +optional
-	Selector *metav1.LabelSelector `json:"selector,omitempty"`
-	// Group is the group of the route resource to select. Omitting the value
-	// indicates the gateway.networking.k8s.io API group.
-	// For example, use the following to select an HTTPRoute:
-	//
-	// routes:
-	//   kind: HTTPRoute
-	//
-	// Otherwise, if an alternative API group is desired, specify the desired
-	// group:
+
+	// Kinds specifies the groups and kinds of Routes that are allowed to bind
+	// to this Gateway Listener. When unspecified or empty, the kinds of Routes
+	// selected are determined using the Listener protocol.
 	//
-	// routes:
-	//   group: acme.io
-	//   kind: FooRoute
+	// A RouteGroupKind MUST correspond to kinds of Routes that are compatible
+	// with the application protocol specified in the Listener's Protocol field.
+	// If an implementation does not support or recognize this resource type, it
+	// SHOULD set the "ResolvedRefs" condition to false for this Listener with
```

Comment:
```suggestion
	// MUST set the "ResolvedRefs" condition to false for this Listener with
```
SHOULD implies that it's not always necessary, I think that MUST is better for status updates.

Reply:
Updated, thanks!